mbed_host_tests/host_tests/base_host_test.py

```diff
@@ -49,6 +49,14 @@ def notify_complete(self, result=None):
         if self.__event_queue:
             self.__event_queue.put(('__notify_complete', result, time()))
 
+    def reset_dut(self):
+        """
+        Reset device under test
+        :return:
+        """
+        if self.__event_queue:
+            self.__event_queue.put(('__reset_dut', True, time()))
```

Comment:
Agreed. 
Currently only SW reset is implemented. In future value will determine SW/HW reset type.